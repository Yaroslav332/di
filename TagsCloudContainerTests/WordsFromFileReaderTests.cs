using System.Collections.Generic;
using System.IO;
using ApprovalTests;
using ApprovalTests.Reporters;
using Autofac;
using NUnit.Framework;
using NUnit.Framework.Internal;
using TagsCloudContainer;

namespace TagsCloudContainerTests;

[TestFixture]
[UseReporter(typeof(DiffReporter))]
public class WordsFromFileReaderTests
{
    private ContainerBuilder builder;
    private string TestFilesDir = Path.Combine(TestContext.CurrentContext.TestDirectory, "TestFiles");

    [SetUp]
    public void Setup()
    {
        builder = new ContainerBuilder();
        builder.RegisterType<WordsFromFileReader>().AsSelf();
    }

    [Test]
    public void WordsFromFileReader_ShouldWorkCorrectly()
    {
        builder.RegisterInstance(Path.Combine(TestFilesDir, "DefaultInput.txt")).As<string>();
        
        using (var scope = builder.Build().BeginLifetimeScope())
        {
            var reader = scope.Resolve<WordsFromFileReader>();
            Approvals.VerifyAll(reader, s => s);
        }


    }
    
    [Test]
    public void WordsFromFileReader_ShouldWorkCorrectly_WhenEmptyFile()
    {
        builder.RegisterInstance(Path.Combine(TestFilesDir, "EmptyInput.txt")).As<string>();
        
        using (var scope = builder.Build().BeginLifetimeScope())
        {
            var reader = scope.Resolve<WordsFromFileReader>();
            Approvals.VerifyAll(reader, s => s);
        }

        

    }
}