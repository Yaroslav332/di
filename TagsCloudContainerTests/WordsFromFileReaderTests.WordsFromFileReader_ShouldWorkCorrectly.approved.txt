A
B
A
A
D
B
